handlers.go

```diff
@@ -356,13 +357,13 @@ func (dht *IpfsDHT) handleAddProvider(ctx context.Context, p peer.ID, pmes *pb.M
 	defer func() { logger.FinishWithErr(ctx, _err) }()
 	logger.SetTag(ctx, "peer", p)
 
-	c, err := cid.Cast([]byte(pmes.GetKey()))
+	_, h, err := mh.MHFromBytes(pmes.GetKey())
```

Comment:
@raulk we'd like to switch from storing CIDs to Multihashes to support finding peers with CIDv0 and CIDv1 of the same multihash. While this will break behavior for current users of CIDv1 since peers with an updated DHT will look for and advertise to the multihash instead of the CID (CIDv0 users are protected since CIDv0 = Multihash SHA256), it will help the network longer term.

On the "server" end we previously checked that the key that we were adding providers for was a CID (the code change above). Initially I started just switching all the things associated with Providers to use Multihashes instead of CIDs. However, this does mean that a new server will reject advertisements from old users of CIDv1.

How would you feel about switching all provider records internals (e.g. `ProviderManager`) to work on `[]byte` instead of on Multihashes or CIDs? We could then have the server check if the key is less than 80 bytes (640 bits) for DoS purposes, but otherwise not care about the content type.

Reply:
I changed this to use `[]byte` instead of Multihashes and to limit the `[]byte` size to 80 bytes. Can revert back if we want.